pkg/middleware/default_http_client.go

```diff
@@ -37,30 +42,63 @@ func DefaultHTTPClient() *http.Client {
 }
 
 func init() {
-	defaultTransport = armbalancer.New(armbalancer.Options{
+	defaultTransport = &http.Transport{
+		Proxy: http.ProxyFromEnvironment,
+		DialContext: (&net.Dialer{
+			Timeout:   30 * time.Second,
+			KeepAlive: 30 * time.Second,
+		}).DialContext,
+		ForceAttemptHTTP2:     true,
+		MaxIdleConns:          100,
+		MaxIdleConnsPerHost:   100,
+		IdleConnTimeout:       90 * time.Second,
+		TLSHandshakeTimeout:   10 * time.Second,
+		ExpectContinueTimeout: 1 * time.Second,
+		TLSClientConfig: &tls.Config{
+			MinVersion: tls.VersionTLS12,
+		},
+	}
```

Comment:
could move this directly in the var declaration as var are instanciated before init.
or can keep it in init with the rest.

I went with keeping it in init()